tokio/src/runtime/time/mod.rs

```diff
@@ -443,24 +474,17 @@ impl Handle {
 // ===== impl Inner =====
 
 impl Inner {
-    /// Locks the driver's sharded wheel structure.
-    pub(super) fn lock_sharded_wheel(
-        &self,
-        shard_id: u32,
-    ) -> crate::loom::sync::MutexGuard<'_, Wheel> {
-        let index = shard_id % (self.wheels.len() as u32);
-        // Safety: This modulo operation ensures that the index is not out of bounds.
-        unsafe { self.wheels.get_unchecked(index as usize).lock() }
-    }
-
     // Check whether the driver has been shutdown
     pub(super) fn is_shutdown(&self) -> bool {
         self.is_shutdown.load(Ordering::SeqCst)
     }
 
     // Gets the number of shards.
     fn get_shard_size(&self) -> u32 {
-        self.wheels.len() as u32
+        self.wheels
+            .read()
+            .expect("Timer wheel shards poisened")
```

Comment:
It's a bit unfortunate that this can panic. Not because the panic  itself is _bad_, if the lock is poisoned, we _should_ panic, but because panicking code will make this  otherwise small and trivially inlined function generate way more code. Perhaps we should  move the number of shards outside of the lock, instead? Since it will never change, it should be fine to do that...

Reply:
also, a typo: 
```suggestion
            .expect("Timer wheel shards poisoned")
```

Reply:
Seems reasonable to put it as a member variable into the `Inner` struct to avoid the lock here. The increase in size of the struct would probably be ok since it's "only" a `u32`, so I would agree to do it like that if nobody objects.

Reply:
I included the changes in  [e9c4351](https://github.com/tokio-rs/tokio/pull/6779/commits/e9c4351901453989f917366ae36c19090238cb6d).

I also removed the method to get the shard size from the struct `ShardedWheel` to prevent duplications.